### PR TITLE
GRPCRoute: Add GRPCRoute support

### PR DESCRIPTION
**Pull request by fangfpeng**
Add kind and cache support for GRPCRoute

Updates #4820

Signed-off-by: Fang Peng <fpeng@vmware.com>

### CODE REVIEW

File: internal/k8s/kind.go [hunk 1]
```diff
@@ -45,6 +46,8 @@ func KindOf(obj interface{}) string {
 			return "HTTPProxy"
 		case *gatewayapi_v1beta1.HTTPRoute:
 			return "HTTPRoute"
+		case *gatewayapi_v1alpha2.GRPCRoute:
+			return "GRPCRoute"
```
Comment (skriss) [inline internal/k8s/kind.go:50]: Will have to look to see exactly where this is used; I'm noticing we don't have TLSRoute here.

**Review by skriss — APPROVED** [review]
LGTM!

File: internal/k8s/kind.go (hunk 1 above)
Comment (fangfpeng) [reply internal/k8s/kind.go:50]: Right, TLSRoute is not there. For routes, KindOf() is used in cache.go, statusaddress.go and metrics.go for some debug/error message or metric counter.

Comment (skriss) [reply internal/k8s/kind.go:50]: :+1: thanks, yeah makes sense to add GRPCRoute here, we should follow up with adding any other missing types for now, and/or see if we can implement this functionality differently such that we don't need to maintain a hardcoded list.

**Pull request merged**